### PR TITLE
add jitsi regional information to all analytics events via permanent properties

### PR DESCRIPTION
**Pull request by aaronkvanmeerten**
add jitsi regional information to all analytics events via permanent properties if available

ignore listing on modules/analytics/analytics.js to avoid es6 linting errors

### CODE REVIEW

File: .jshintignore [hunk 1]
```diff
@@ -16,6 +16,7 @@ react/
 # The following are checked by ESLint with the minimum configuration which does
 # not supersede JSHint but take advantage of advanced language features such as
 # Facebook Flow which are not supported by JSHint.
+modules/analytics/analytics.js
```
Comment (paweldomas) [inline .jshintignore]: maybe it would be better to disable particular rule using inline comments instead of ignoring the whole file ?
example:
/* eslint-disable some-rule */
some ugly hacky code
/* eslint-enable some-rule */

Comment (aaronkvanmeerten) [inline .jshintignore]: Good question.   The issue is with jshint, which doesn't understand the newer es6 syntax and therefore isn't going to properly lint this file.  Lyubo suggested that we simply ignore it entirely since jshint will be going away eventually anyway.

File: modules/analytics/analytics.js [hunk 2]
```diff
@@ -111,7 +111,8 @@ class Analytics {
             handlers => {
                 const permanentProperties = {
                     roomName: APP.conference.roomName,
-                    userAgent: navigator.userAgent
+                    userAgent: navigator.userAgent,
+                    ...window.jitsiRegionInfo
```
Comment (paweldomas) [inline modules/analytics/analytics.js]: I think I would prefer explicit declaration/copy of each attribute instead of copying whole object, which type is not defined/described anywhere

Comment (aaronkvanmeerten) [inline modules/analytics/analytics.js]: I have refactored this PR to explicitly extract the expected properties.

**Comment by aaronkvanmeerten** [comment]
I have changed my PR to reflect all comments.

File: modules/analytics/analytics.js [hunk 3]
```diff
@@ -111,7 +111,7 @@ class Analytics {
             handlers => {
                 const permanentProperties = {
                     roomName: APP.conference.roomName,
-                    userAgent: navigator.userAgent
+                    userAgent: navigator.userAgent,
```
Comment (saghul) [inline modules/analytics/analytics.js]: uneeded

File: modules/analytics/analytics.js [hunk 4]
```diff
@@ -123,6 +123,13 @@ class Analytics {
                     permanentProperties.group = group;
                 }
 
+                if (window.jitsiRegionInfo) {
```
Comment (saghul) [inline modules/analytics/analytics.js]: Can we put this into `window.JitsiMeetJS.regionInfo` instead? A little while ago we settled on that as our only global object (we are not there yet, but new code should follow the rules! 👮 ) See: https://github.com/jitsi/jitsi-meet/blob/master/modules/util/helpers.js#L50

File: modules/analytics/analytics.js [hunk 5]
```diff
@@ -123,6 +123,13 @@ class Analytics {
...
+                    permanentProperties.ProxyRegion;
```
Comment (paweldomas) [inline modules/analytics/analytics.js]: `permanentProperties.ProxyRegion;`
I don't get how is this supposed to work ?
Is it one of some magic expressions ?

File: modules/analytics/analytics.js (hunk 4 above)
Comment (aaronkvanmeerten) [inline modules/analytics/analytics.js]: This would involve changes to the rest of the codebase where these values are fetched and created.  If this is a worthwhile change I am willing to make it across the whole codebase, but I don't think it belongs in this PR.  This values are defined in external_connect.js and then referenced in a few other places in the code.

File: modules/analytics/analytics.js (hunk 5 above)
Comment (paweldomas) [inline modules/analytics/analytics.js]: I would expect:
`permamentPropeties.Region = window.JitsiMeetJS.regionInfo.Region`
Am I missing something ?

File: modules/analytics/analytics.js (hunk 4 above)
Comment (paweldomas) [inline modules/analytics/analytics.js]: @saghul thanks for letting us know that such agreement exists. I will propose on next retro that we add such agreements to our coding style .md file into "special agreements" section or something.

Comment (saghul) [inline modules/analytics/analytics.js]: The agreement is about not having global variables in new code, pretty much. Now, I understand this is not new code, so we can let this one slide and refactor things together with the Electron desktop picker, for example.

Comment (aaronkvanmeerten) [inline modules/analytics/analytics.js]: Agreed, global variables should be worked on in another PR, which will include work on lib-jitsi-meet as well.

Comment (hristoterezov) [inline modules/analytics/analytics.js]: @paweldomas +1 for the proposal about writing down the "special agreements" in the coding style .md. Great idea! If we don't start doing this probably most of the agreements will be forgotten!

**Comment by aaronkvanmeerten** [comment]
OK this one can be removed now, I have a new PR
